lib/portage/dbapi/vartree.py

```diff
@@ -4867,64 +4876,61 @@ def onerror(e):
 
                     for pkg in owners:
                         pkg = self.vartree.dbapi._pkg_str(pkg.mycpv, None)
-                        pkg_info_str = "%s%s%s" % (pkg, _slot_separator, pkg.slot)
+                        pkg_info_str = f"{pkg}{_slot_separator}{pkg.slot}"
                         if pkg.repo != _unknown_repo:
-                            pkg_info_str += "%s%s" % (_repo_separator, pkg.repo)
+                            pkg_info_str = f"{pkg_info_str}{_repo_separator}{pkg.repo}"
                         pkg_info_strs[pkg] = pkg_info_str
 
                 finally:
                     self.unlockdb()
 
                 for pkg, owned_files in owners.items():
-                    msg = []
-                    msg.append(pkg_info_strs[pkg.mycpv])
-                    for f in sorted(owned_files):
-                        msg.append(
-                            "\t%s" % os.path.join(destroot, f.lstrip(os.path.sep))
-                        )
-                    msg.append("")
+                    msg = [
+                        pkg_info_strs[pkg.mycpv],
+                        *(
+                            f"\t{os.path.join(destroot, f.lstrip(os.path.sep))}"
+                            for f in sorted(owned_files)
+                        ),
+                        "",
+                    ]
                     eerror(msg)
 
                 if not owners:
                     eerror(
                         [_("None of the installed" " packages claim the file(s)."), ""]
                     )
 
-            symlink_abort_msg = _(
-                "Package '%s' NOT merged since it has "
-                "one or more collisions between symlinks and directories, "
-                "which is explicitly forbidden by PMS section 13.4 "
-                "(see bug #326685)."
-            )
-
             # The explanation about the collision and how to solve
             # it may not be visible via a scrollback buffer, especially
             # if the number of file collisions is large. Therefore,
             # show a summary at the end.
             abort = False
             if symlink_collisions:
                 abort = True
-                msg = symlink_abort_msg % (self.settings.mycpv,)
+                msg = _(
+                    f"Package '{self.settings.mycpv}' NOT merged since it has "
+                    "one or more collisions between symlinks and directories, "
+                    "which is explicitly forbidden by PMS section 13.4 "
+                    "(see bug #326685)."
+                )
             elif collision_protect:
```

Comment:
I suppose you could combine this and the next condition and OR them together as they have the same effect, like you've done elsewhere.

Reply:
Another nice catch! It's appreciated!